cpp/src/groupby/sort/group_tdigest.cu

```diff
@@ -408,13 +422,100 @@ generate_group_cluster_info(int delta,
     cumulative_weight,
     group_cluster_wl.begin(),
     group_num_clusters.begin(),
-    group_cluster_offsets->view().begin<offset_type>());
+    group_cluster_offsets->view().begin<offset_type>(),
+    has_nulls);
 
   return {std::move(group_cluster_wl),
           std::move(group_cluster_offsets),
           static_cast<size_type>(total_clusters)};
 }
 
+std::unique_ptr<column> build_output_column(size_type num_rows,
+                                            std::unique_ptr<column>&& means,
+                                            std::unique_ptr<column>&& weights,
+                                            std::unique_ptr<column>&& offsets,
+                                            std::unique_ptr<column>&& min_col,
+                                            std::unique_ptr<column>&& max_col,
+                                            bool has_nulls,
+                                            rmm::cuda_stream_view stream,
+                                            rmm::mr::device_memory_resource* mr)
+{
+  // whether or not this weight is a stub
+  auto is_stub_weight = [weights = weights->view().begin<double>()] __device__(size_type i) {
+    return weights[i] == 0;
+  };
+  // whether or not this particular tdigest is a stub
+  auto is_stub_digest = [offsets = offsets->view().begin<offset_type>(), is_stub_weight] __device__(
+                          size_type i) { return is_stub_weight(offsets[i]) ? 1 : 0; };
+
+  size_type const num_stubs = [&]() {
+    if (!has_nulls) { return 0; }
+    auto iter = cudf::detail::make_counting_transform_iterator(0, is_stub_digest);
+    return thrust::reduce(rmm::exec_policy(stream), iter, iter + num_rows);
+  }();
+
+  // if there are no stub tdigests, we can return immediately.
+  if (num_stubs == 0) {
+    return cudf::detail::tdigest::make_tdigest_column(num_rows,
+                                                      std::move(means),
+                                                      std::move(weights),
+                                                      std::move(offsets),
+                                                      std::move(min_col),
+                                                      std::move(max_col),
+                                                      stream,
+                                                      mr);
+  }
+
+  // otherwise we need to strip out the stubs.
+
+  // remove from the means and weights column
+  auto _means = cudf::make_fixed_width_column(
```

Comment:
Nitpick: Might we consider `cudf::make_numeric_column()` to skip some of the checks in `make_fixed_width_column()`?
```suggestion
  auto _means = cudf::make_numeric_column(
```